### ISSUE TITLE
Take picture crash - Android 

### ISSUE THREAD

**Issue by ngt14** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#360]
Hello, i recently update plugin to resolved **"Added fix for rotating images based on exif data"** #355, it resolved on my Huawei but on my samsung devices, my app crash when i take picture.

![capture d ecran 2017-09-12 a 14 31 28](https://user-images.githubusercontent.com/18715353/30326205-097d56d4-97c8-11e7-9e1c-5531b56c81da.png)

what is the problem exactly?

**Comment by rodreiser** [comment]
same for me when testing on samsung galaxy s6, with android version 5.1

**Comment by bvx89** [comment]
Hi,

I was the one who introduced this functionality, but I didn't check if the ExifInterface method existed on earlier versions of Android. I have since then included a reference to the support library which includes the missing ExifInterface method, so hopefully this is fixed in the latest master.

Please let me know if it's still not working.

Sorry for the inconvenience!

**Comment by jedofus** [comment]
Hello @bvx89 , I have just try this and now i can't build my app. 
I have this error : 
![capture d ecran 2017-09-15 a 14 56 31](https://user-images.githubusercontent.com/18715353/30483798-80e81d50-9a28-11e7-9cf7-fdf9e7c19eff.png)

**Comment by bvx89** [comment]
Yeah, I was experiencing this error as well. Could you try to clone this project and add it to your cordova project from the local file system? In the `config.xml`-file at the root of the project, there's a line inside the android platform configuration that says `<framework src="com.android.support:exifinterface:25+" />`. If you try to change this before you added it to your cordova app so that it says `<framework src="com.android.support:exifinterface:26" />`, it should hopefully make the errors go away. 

Alternatively, you can try to remove the line in the config and see if that works.

Sadly, I don't know very much about cordova plugins and how to configure them, but I tried to contribute with the code changes that worked for our projects.

**Comment by westonganger** [comment]
@jedofus it looks like you are trying to build for Android 8 (Android API level 26). Cordova only supports up to API level 25 right now.  See https://cordova.apache.org/docs/en/latest/guide/platforms/android

**Comment by westonganger** [comment]
This should now be fixed in master. Please update to master and let me know if its functioning.

**Comment by ngt14** [comment]
Hi, 
it's work now, thanks !

**Comment by trigoesrodrigo** [comment]
I'm still seeing this on a HTC m8 with Android 6.0

**Comment by bvx89** [comment]
@trigoesrodrigo The bug is happening on all Android versions less than API level 24 (Nougat). Android 6.0 (Marshmellow) is using API level 23. The code that is in master now uses a constructor from ExifInterface that reads the exif data from a InputStream, which was made available in API level 24.

You can see this in the [android developer reference](https://developer.android.com/reference/android/media/ExifInterface.html#pubctors) for the ExifInterface constructor by changing the API-version level to 23.

@westonganger My suggestion is that you undo the commit [eabc693367940fe47b9044c0eb3420eca3f64ac2](https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/commit/eabc693367940fe47b9044c0eb3420eca3f64ac2), and instead find a way to make the support library get installed without the problem that @ngt14 experienced.

**Comment by westonganger** [comment]
Sure. Can you determine what the valid framework tag would be?
 Please create a PR with these changes and I'll merge it.

**Comment by ngt14** [comment]
Have you found a way to solve this problem?

**Comment by sertal70** [comment]
I'm experiencing this issue, too. It's breaking my app on all android devices older than 7 and, you know, there are quite a lot out there. 

I've read all comments and it seems that the root cause it the fix #355 made by @bvx89 and merged one month ago by @westonganger. 

@bvx89 if you can patch your fix, please do it as soon as possibile. Otherwise, please @westonganger can you revert the master to allow the plugin work again in Android <7 ?

Thanks!

**Comment by sertal70** [comment]
@bvx89 do you really think your proposal is a fix? I do not. I think that the state of the plugin should allow any user to download, compile and run it, as simple as that. If anyone starts to introduce "fixes" that require manual edits on configuration files, fate of the plugin is sealed.

Your fix #355 has broken compatibility with more than 60% of the android market, my opinion is that  it should be retired until you can find a better solution and this is my request to @westonganger.

**Comment by bvx89** [comment]
@sertal70 I never said it was a fix, I was merely suggestion what you could do until the problem is solved. I agree 100% that it should be as easy as including the plugin and run it, but I do not know how. The issue about library versions is also an issue that [other cordova plugins](https://github.com/phonegap/phonegap-plugin-push/issues/1842#issuecomment-313979240) have.

Sure, my commit could be removed, but then you end up with photos that might be rotated, and you have no way of knowing if they are rotated (since It only happens on certain devices, mostly Samsung devices from our internal testing). You also have no idea of knowing which way they are rotated either, so you can't automatically rotate the image later with the base64-string to get the correct orientation. Considering what I wrote above, Isn't omitting my fix kind of breaking the compatibility with a percent of the market as well?

In my perspective, taking the extra step of making the app build with one of my suggested fixes above (a one-time setup for the app) is a small price to pay for getting consistent image results from the plugin. At least while waiting for a proper fix.

**Comment by sertal70** [comment]
@bvx89 it is clear we have different opinion... 

#355 solve the orientation issue but imho it comes to a too high price in terms of compatibility. I surely  appreciate your work but, to be complete it should avoid to break compatibility with such a large number of devices.

Said that, it is unclear to me what should be changed in the `platforms/android/project.properties` file.  My file content is as follows:

```
target=android-25
android.library.reference.1=CordovaLib
cordova.system.library.1=com.android.support:support-v4:25.+
cordova.system.library.2=com.android.support:appcompat-v7:25.+
cordova.system.library.3=com.android.support:support-v13:25.1.0
cordova.system.library.4=me.leolin:ShortcutBadger:1.1.17@aar
cordova.system.library.5=com.google.firebase:firebase-messaging:11.0.1
cordova.gradle.include.1=phonegap-plugin-push/sot-push.gradle

```

which line should be changed and how?

**Comment by westonganger** [comment]
If you upgrade to cordova android 6.3.0 maybe that will fix this as they
updated to api level 26.

**Comment by bvx89** [comment]
@sertal70 Could you post your error message? Are you getting the error when building or when running your app?
